config/crd/patches/max_name_size_for_assign.yaml

```diff
@@ -1,13 +1,200 @@
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   name: assign.mutations.gatekeeper.sh
 spec:
-  validation:
-    openAPIV3Schema:
-      properties:
-        metadata:
-          properties:
-            name:
-              type: string
-              maxLength: 63
\ No newline at end of file
+  versions:
+  - name: v1alpha1
```

Comment:
These massive patches are unfortunately the best I could do given the new structure of the v1 CRD.

TL;DR: `strategicMergePatch` isn't properly supported in CRDs, and CRDs changed their shape from v1beta1 to v1

More context:

The first relevant detail here is that, in v1, CRDs did away with the single-schema format of the CRD that we saw in `v1beta1`.  I.e. in a `v1` CRD, each version's schema is included in its entirety.  These versions (even if there is only one) are included in an array (`versions:`).  

For most kubernetes types that include an array (like a `Deployment`), a specific key is defined as the "map key".  This map key allows one to select an element from the array as if it were a map.  One subsection of the CRD status [actually has this](https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/v1/types.go#L333).

But because a CRD is user-defined, the API server ingests it as an unstructured.  This prevents it from implementing this pre-baked mapkey.

In the previous version of this patch yaml, we were operating on a v1beta1 CRD, which didn't have to deal with an array.  Thus, the simple patch worked.  But because we now have to deal with an array and that array lacks the internal mapkey structure we'd expect, we have to _replace the entire array item_.

This is obviously less than ideal.

But, perhaps this will go away as we get mutation out of alpha?

This was blocking the v1 CRD change, which is an existential requirement for g8r going forward.  I felt that the compromise was appropriate.

LMK if you see a better way around this!

Reply:
Resolved this with `patchJson6902`.  Worked like a charm and was much shorter 🤟🏼